internal/legacy/pinhal_baremetal.go

```diff
@@ -0,0 +1,33 @@
+//go:build baremetal
+
+package legacy
+
+import "machine"
+
+func configurePinOut(p PinOutput) {
+	configurePin(p, machine.PinOutput)
+}
+
+func configurePinInputPulldown(p PinInput) {
+	configurePin(p, pulldown) // some chips do not have pull down, in which case pulldown==machine.PinInput.
+}
+
+func configurePinInput(p PinInput) {
+	configurePin(p, machine.PinInput)
+}
+
+func configurePinInputPullup(p PinInput) {
+	configurePin(p, pullup) // some chips do not have pull up, in which case pullup==machine.PinInput.
+}
+
+func pinIsNoPin(a any) bool {
```

Comment:
Is this function really needed? Mostly it is better to call a function and return an error, if the condition is not met at this time than check the condition some time before and call the function later if the condition was met in the past. Reason: the state can change between check and call, which leads to sporadic errors.